rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -324,6 +324,20 @@ def default_call_order(self, skip_etherscan: bool = False) -> list[WeightedNode]
             ordered_list = [WeightedNode(node_info=node, weight=ONE, active=True) for node in owned_nodes] + ordered_list  # noqa: E501
         return ordered_list
 
+    def archive_nodes_call_order(self) -> list[WeightedNode]:
+        """Orders nodes by placing archive nodes first, followed by non-archive nodes"""
+        call_order = self.default_call_order()
+        ordered_list: list[WeightedNode] = []
+        for node in call_order:
+            if node.node_info in self.web3_mapping:
+                if self.web3_mapping[node.node_info].is_archive:
+                    ordered_list.insert(0, node)
+                else:
+                    ordered_list.append(node)
+            else:
+                ordered_list.append(node)
+        return call_order
```

Comment:
probably you will remove this logic as it is but think deeper your code

```suggestion
        for node in call_order:
            if (
                node.node_info in self.web3_mapping:
                self.web3_mapping[node.node_info].is_archive
             ):
                    ordered_list.insert(0, node)
             else:
                    ordered_list.append(node)

        return call_order
```